cmd/gapir/cc/main.cpp

```diff
@@ -63,20 +67,84 @@ std::vector<uint32_t> memorySizes {
       128 * 1024 * 1024U,       // 128MB
 };
 
-// createResourceProvider constructs and returns a ResourceInMemoryCache.
+#if TARGET_OS == GAPID_OS_LINUX
+std::string getOnDiskCacheFilesPath(const std::string parentPath) {
+  std::string t("/gapir-cache.XXXXXX");
+  if (parentPath.size() == 0) {
+    const char* tmpDir = std::getenv("TMPDIR");
+    if (tmpDir == nullptr) {
+      return "";
+    }
+    t = std::string(tmpDir) + t;
+  } else {
+    std::string p(parentPath);
+    while (p.back() == '/') {
+      p.pop_back();
+    }
+    t = p + t;
+  }
+  std::vector<char> v(t.begin(), t.end());
+  v.push_back('\0');
+  char* path = mkdtemp(v.data());
+  if (path == nullptr) {
+    return "";
+  }
+  return path;
+}
+#endif
+
+// createResourceLoader constructs and returns a ResourceInMemoryCache.
 // If cachePath is non-null then the ResourceInMemoryCache will be backed by a
 // disk-cache.
-std::unique_ptr<ResourceInMemoryCache> createResourceProvider(
-    const char* cachePath, MemoryManager* memoryManager) {
-  if (cachePath != nullptr) {
-    GAPID_FATAL("Disk cache is currently out of service. Got %s", cachePath);
-    return std::unique_ptr<ResourceInMemoryCache>(ResourceInMemoryCache::create(
-        ResourceDiskCache::create(ResourceRequester::create(), cachePath),
-        memoryManager->getBaseAddress()));
-  } else {
-    return std::unique_ptr<ResourceInMemoryCache>(ResourceInMemoryCache::create(
-        ResourceRequester::create(), memoryManager->getBaseAddress()));
+std::unique_ptr<ResourceCache> createCache(bool enableDiskCache,
+                                           const std::string& diskCachePath,
+                                           MemoryManager* memoryManager) {
+#if TARGET_OS == GAPID_OS_LINUX
+  std::string cacheFilesPath;
+  if (enableDiskCache) {
+    cacheFilesPath = getOnDiskCacheFilesPath(diskCachePath);
+    if (cacheFilesPath.size() == 0) {
+      GAPID_WARNING(
+          "No disk cache path specified and no $TMPDIR environment variable "
+          "defined, fallback to disable on-disk cache.");
+      enableDiskCache = false;
+    }
   }
+  if (enableDiskCache) {
+    if (fork() == 0) {
```

Comment:
[Something relevant I was reading this morning](http://www.linuxprogrammingblog.com/threads-and-fork-think-twice-before-using-them). GAPIR does use threads - so we might need to be careful about `fork` here.

Reply:
I think this specific case we are ok. This is called from startServer() which happens before we start up GRPC. So I don't think we will have any extra threads running.